language/move-prover/bytecode-to-boogie/src/lifetime_analysis.rs

```diff
@@ -0,0 +1,361 @@
+// Copyright (c) The Libra Core Contributors
+// SPDX-License-Identifier: Apache-2.0
+
+use crate::{
+    dataflow_analysis::{DataflowAnalysis, StateMap, TransferFunctions},
+    stackless_control_flow_graph::StacklessControlFlowGraph,
+};
+use bytecode_verifier::absint::{AbstractDomain, JoinResult};
+use stackless_bytecode_generator::stackless_bytecode::StacklessBytecode::{self, *};
+use std::collections::{BTreeMap, BTreeSet};
+use vm::file_format::{CodeOffset, LocalIndex, SignatureToken};
+
+#[derive(Clone, Copy, Debug, Eq, Ord, PartialEq, PartialOrd)]
+pub enum EdgeType {
+    Local,
+    Global,
+    Field,
+}
+
+#[derive(Clone, Copy, Debug, Eq, Ord, PartialEq, PartialOrd)]
+pub enum Node {
+    Root,
+    Local(LocalIndex),
+}
+
+#[derive(Clone, Copy, Debug, Eq, Ord, PartialEq, PartialOrd)]
+pub struct Edge {
+    edge_type: EdgeType,
+    to: Node,
+}
+
+#[derive(Clone, Debug, Default, Eq, Ord, PartialEq, PartialOrd)]
```

Comment:
Turns out I do need the clone unfortunately because it is part of the State, which in the dataflow analysis framework is cloned a few times in initialization. Maybe there's a better way?

Reply:
I was mostly concerned about Ordering because this may generate quite some code.